core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -2903,9 +2928,30 @@ private boolean isUniqueUserIdEnabled(UserStoreManager userManager) {
         return userManager instanceof AbstractUserStoreManager && ((AbstractUserStoreManager) userManager).isUniqueUserIdEnabled();
     }
 
+    /**
+     * This is to check whether the unique group id is enabled for the userstore.
+     *
+     * @param userManager UserStoreManager/
+     * @return True if unique group id is enabled for the userstore.
+     */
+    private boolean isUniqueGroupIdEnabled(UserStoreManager userManager) {
+
+        if (!(userManager instanceof AbstractUserStoreManager)) {
+            return false;
+        }
+        // todo implement group uuid support for following UserStoreManagers.
```

Comment:
shall we create a git issue to track this improvement and remove TODOs?

Reply:
We already have this captured as an issue in the initial tasks list. I will add this in the comment as well.

https://github.com/wso2/product-is/issues/7914